platform-core-module/src/main/java/com/softicar/platform/core/module/file/smb/ISmbDirectory.java

```diff
@@ -15,20 +16,99 @@
 	 */
 	void mkdirs();
 
+	/**
+	 * Lists all files in this directory.
+	 *
+	 * @return the files in this directory (never <i>null</i>)
+	 */
 	List<ISmbFile> listFiles();
 
-	Collection<String> listFilesRecursively();
+	/**
+	 * Lists the files in this directory and all of its sub-directories.
+	 *
+	 * @return the files in this directory and its sub-directories (never
+	 *         <i>null</i>)
+	 */
+	List<ISmbFile> listFilesRecursively();
 
-	ISmbFile getFile(String name);
+	/**
+	 * Lists all sub-directories in this directory.
+	 *
+	 * @return the sub-directories in this directory (never <i>null</i>)
+	 */
+	List<ISmbDirectory> listSubDirectories();
```

Comment:
because it's "subdirectories" -> listSubdirectories()

Reply:
method name is still `listSubDirectories()`

Reply:
I personally veto this, `listSubDirectories` has way better readability

Reply:
Actually I missed this - only renamed `getSubdirectory`.
However, I now reverted this renaming because `SubDirectory`:
- has superior legibility
- reflects the returned type (`ISmb**Directory**`).
In javadoc, "subdirectory" remains, of course.
I hope that this is an acceptable compromise.

Reply:
`Sub` infix was removed.